javascript/javascript3/week1/weatherApp/script.js

```diff
@@ -0,0 +1,109 @@
+// @ts-nocheck
+window.addEventListener('load', () => {
+  let lat;
+  let long;
+  let geoBtn = document.querySelector('.geoBtn');
+  geoBtn.addEventListener('click', function getLocation() {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        let lat = position.coords.latitude.toFixed(2); //to fixed is decimal
+        let long = position.coords.longitude.toFixed(2);
+        const text = document.querySelector('.lat');
+        text.innerHTML = `Latitude is ${lat}`;
+        const text2 = document.querySelector('.long');
+        text2.innerHTML = `Longitude is ${long}`;
+      });
+    } else {
+      alert('Cannot access your location');
+    }
+  });
+
+  const button = document.querySelector('.searchBtn');
+  const inputText = document.getElementById('inputText');
+  const locationPlace = document.querySelector('.location-place');
+  const icon = document.querySelector('.icon');
+  const description = document.querySelector('.temperature-description');
+  const humidityTemp = document.querySelector('.description-humidity');
+  const wind = document.querySelector('.description-wind');
+  const pressure = document.querySelector('.description-pressure');
+  const coordLat = document.querySelector('.lat');
+  const coordLong = document.querySelector('.long');
+
+  button.addEventListener('click', () => {
+    fetch(
+      'https://api.openweathermap.org/data/2.5/weather?q=' +
+        inputText.value +
+        '&appid=99f4e75ba68210ceb835bd5467615b98'
+    )
+      .then((response) => response.json())
+      .then((data) => {
```

Comment:
What happends if the user provides a wrong city and you get nothing from the weather API?
So before starting to handle the data, you should check if there is actually some data, and if there is nothing, tell the user that the provided location does was not found, or its wrong....